### DISCUSSION TITLE
Feature Request: background fetch OTA updates

### DISCUSSION THREAD

**Discussion by nandorojo** [expo/expo#19813] [Proposals and Ideas]
@EvanBacon and Charlie seemed to think background fetching OTA updates was very doable.

Currently, EAS updates only fetch when users open the app. This isn't ideal since it requires one of the following options:

1. Users have to see a splash / loading screen for a long time while they wait for their update, or
2. You have to serve users a stale app and only apply the update on the next session.

For BeatGig, we opted for option 1, since we want users to always have the latest version. This option reflects the web experience too.

Fetching OTA bundles in the background would make this much better. 

With all the hype around Hermes and fast app open times, it's worth noting that OTA updates that block users from entering the app is our biggest hurdle for good TTI.

**Comment by sregg** [comment]
Has anyone managed to implement some sort of background OTA download?

**Reply by XahidEx** [reply]
I was able to do that. But it's not entirely stable. Usually it takes 2-4 seconds to update (in the background). Rarely it takes hours.

**Comment by diegoalfonsoc18** [comment]
We faced the same dilemma with our production app. We went with option 2 (serve stale + apply on next session) but improved it using expo-updates hooks:
import * as Updates from 'expo-updates';

// Check for updates silently in the background after app loads
useEffect(() => {
  async function checkForUpdates() {
    try {
      const update = await Updates.checkForUpdateAsync();
      if (update.isAvailable) {
        await Updates.fetchUpdateAsync();
        // Optionally prompt user or reload on next app backgrounding
      }
    } catch (e) {
      // Silent fail - user keeps using current version
    }
  }
  checkForUpdates();
}, []);
This way the user enters the app immediately (no splash delay), the update downloads silently, and it applies on the next cold start. Not perfect, but a solid middle ground until true background fetch is supported.

That said, real background fetching via iOS Background App Refresh / Android WorkManager would be a game changer. The update would already be downloaded before the user even opens the app — zero TTI impact.
